### PR TITLE
docker-machine-driver-xhyve: fix libev issue

### LINKED ISSUE #6678: docker-machine-driver-xhyve failed to build on 10.10.5
### Bug reports:

New version of `docker-machine-driver-xhyve`, `0.3.0`, fails to build with the following message:

```
==> make build
Last 15 lines from /Users/rag/Library/Logs/Homebrew/docker-machine-driver-xhyve/04.make:
  "_ev_loop_destroy", referenced from:
      _lwt_libev_stop in liblwt-unix_stubs.a(lwt_libev_stubs.o)
  "_ev_loop_new", referenced from:
      _lwt_libev_init in liblwt-unix_stubs.a(lwt_libev_stubs.o)
  "_ev_run", referenced from:
      _lwt_libev_loop in liblwt-unix_stubs.a(lwt_libev_stubs.o)
  "_ev_set_invoke_pending_cb", referenced from:
      _lwt_libev_init in liblwt-unix_stubs.a(lwt_libev_stubs.o)
  "_ev_timer_start", referenced from:
      _lwt_libev_timer_init in liblwt-unix_stubs.a(lwt_libev_stubs.o)
  "_ev_timer_stop", referenced from:
      _lwt_libev_timer_stop in liblwt-unix_stubs.a(lwt_libev_stubs.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [bin/docker-machine-driver-xhyve] Error 2
```

https://gist.github.com/mtpereira/488d34625ae4639812d0c21aa41bd1cc

### PR DESCRIPTION
**Pull request by zchee**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Change to without `make` build.
`libev` static library file was hardcoded my `Makefile`, It is not a good idea to update the version for fix it.
So, I wrote the doing with the `Makefile`.

Close #6678

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
> So, I wrote the doing without the Makefile.

Could the Makefile be patched instead? Then formula is pretty messy after this. Ideally if you are going to or have fixed this upstream already we'll just apply the upstream patch.

**Comment by zchee** [comment]
@MikeMcQuaid I see. will do.

BTW, What command is can be tested efficiently? `brew audit` is only lint(IIRC), What kind of command is good for install test in the **sandbox**?
I saw `man brew`, but `install` command haven't `--dry-run` flag.
I build on my repository directory with `make`, always use the latest dev(because I'm an owner). 
So I do not want to install by `brew`

**Comment by MikeMcQuaid** [comment]
@zchee You could use `brew test-bot` in CI to test installation is working correctly. I'm not quite sure what you're asking, though, could you rephase/elaborate? Thanks!

**Comment by zchee** [comment]
@MikeMcQuaid Fixed with `patch :DATA`. PTAL.

**Comment by zchee** [comment]
@MikeMcQuaid 
> You could use brew test-bot in CI to test installation is working correctly. I'm not quite sure what you're asking, though, could you rephase/elaborate? Thanks!

Thanks advice :)

File: Formula/docker-machine-driver-xhyve.rb [hunk 1]
```diff
@@ -38,8 +40,12 @@ def install
 
       if build.with? "qcow2"
         build_tags << " qcow2"
+        ENV["LIBEV_FILE"] = "#{Formula["libev"].lib}/libev.a"
```
Comment (MikeMcQuaid) [inline Formula/docker-machine-driver-xhyve.rb]: If you're using the static library you can do `depends_on "libev" => :build` above.

File: Formula/docker-machine-driver-xhyve.rb [hunk 2]
```diff
@@ -23,6 +23,8 @@ class DockerMachineDriverXhyve < Formula
     depends_on "libev"
   end
 
+  patch :DATA
```
Comment (MikeMcQuaid) [inline Formula/docker-machine-driver-xhyve.rb]: Can you please add a link to the upstream patch commit and explanation of why the patch is needed in a comment in the formula file. Thanks!

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/docker-machine-driver-xhyve.rb (hunk 2 above)
Comment (zchee) [inline Formula/docker-machine-driver-xhyve.rb]: That's patch is not merge to upstream yet. Because `$(LIBEV_FILE)` is just carried from
https://github.com/docker/hyperkit/blob/master/Makefile#L83
I would like to follow hyperkit as much as possible. but, Now I'm considering allowing the user to specify the path. So, there is no commit url yet.

Also, sometimes Go users set `$GO_LDFLAGS` to shell global. i.e. always strip symbol table & DWARF with `-s -w`.
In this case, if change the `+=` to `?=`, GNU Make handling shell `$GO_LDFLAGS` env(IIRC).
It means, ignores `-X=xhyve.GitCommit=(...)` trick.
There is a possibility that I can not take a commit hash on the issue, So I am not going to change upstream.

Well, If some comment is necessary, I would appreciate it if I could get your advice, because I am not good at English...

File: Formula/docker-machine-driver-xhyve.rb (hunk 1 above)
Comment (zchee) [inline Formula/docker-machine-driver-xhyve.rb]: Wait for the above reply, and fix this together.

File: Formula/docker-machine-driver-xhyve.rb (hunk 2 above)
Comment (zchee) [inline Formula/docker-machine-driver-xhyve.rb]: Ah, In other words, I will not this patch to my repository, because have side effects.
This patch is Homebrew specific.

I'll add `# Allow specifying version and libev location`.

**Comment by MikeMcQuaid** [comment]
> Ah, In other words, I will not this patch to my repository, because have side effects.
> This patch is Homebrew specific.
> 
> I'll add # Allow specifying version and libev location.

Just to pull this out so it's visible: Homebrew does not accept Homebrew-specific patches that will never be merged upstream. We'll need to figure out something that can work both for your upstream project and Homebrew.

**Comment by zchee** [comment]
@MikeMcQuaid Hmm...
Do you also not admit `inreplace`?

**Comment by MikeMcQuaid** [comment]
> Do you also not admit inreplace?

@zchee We'd pretty strongly rather avoid it. It makes it much harder to keep the formula up to date. What's wrong with your patch being merged upstream as-is?

**Comment by zchee** [comment]
@MikeMcQuaid 
> What's wrong with your patch being merged upstream as-is?

As I said on https://github.com/Homebrew/homebrew-core/pull/6875#discussion_r88623112, below are why I do not merge patches (after the `__END__` in `docker-machine-driver-xhyve.rb`) to my repository.

> $(LIBEV_FILE) is just carried from
https://github.com/docker/hyperkit/blob/master/Makefile#L83
I would like to follow hyperkit as much as possible.

> Also, sometimes Go users set `$GO_LDFLAGS` to shell global. i.e. always strip symbol table & DWARF with `-s -w`.
In this case, if change the `+=` to `?=`, GNU Make handling shell `$GO_LDFLAGS` env(IIRC).
It means, ignores `-X=xhyve.GitCommit=(...)` trick.
There is a possibility that I can not take a commit hash on the issue, So I am not going to change upstream.

In other words, without `brew`, If some users `export GO_LDFLAGS='-s -w'` and `make`, this variable will not change to commit hash. always `HEAD`.
https://github.com/zchee/docker-machine-driver-xhyve/blob/master/xhyve/version.go#L8
So, I will not be able to determine the binary's *commit level* version.

This patch have problems without `brew`.
The `brew` command is executed within the sandbox and overwrite `$GO_LDFLAGS`, so there is no problem. but it is not good for self-build users.

If following the tool manager's policy, and change my repository's `Makefile` just for `brew`, it may be difficult to debug an issue.
I can not consent to it.

**Comment by MikeMcQuaid** [comment]
> In other words, without brew, If some users export GO_LDFLAGS='-s -w' and make, this variable will not change to commit hash. always HEAD.

You could instead provide a `PACKAGER` variable or similar which adds to the version rather than overriding the `GO_LDFLAGS` as a whole.

> If following the tool manager's policy, and change my repository's Makefile just for brew, it may be difficult to debug an issue.

Providing options for packagers is pretty well established as a practice.

> $(LIBEV_FILE) is just carried from
> https://github.com/docker/hyperkit/blob/master/Makefile#L83
> I would like to follow hyperkit as much as possible.

Allowing to to be overridden seems acceptable and I suspect upstream `hyperkit` would accept that too. If it isn't, you could remove the `qcow2` option entirely to remove one patch from here.

**Comment by MikeMcQuaid** [comment]
@zchee What's the latest on this?

**Comment by zchee** [comment]
@MikeMcQuaid I'll release v0.3.1 soon, will fix together.

**Comment by MikeMcQuaid** [comment]
@zchee Great, thanks.
